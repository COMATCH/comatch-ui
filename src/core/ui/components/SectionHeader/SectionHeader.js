import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './SectionHeader.scss';

const propTypes = {
    addOn: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
    className: PropTypes.string,
    size: PropTypes.oneOf(['xs', 'sm', 'xl']),
    title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
};

const defaultProps = {
    addOn: null,
    children: [],
    className: null,
    size: 'xl',
    title: '',
};

function getHeading(size, title, addOn) {
    switch (size) {
        case 'xs':
            return (
                <span>
                    <h6 className="SectionHeader__title">{title}</h6>
                    {addOn && <span className="SectionHeader__addOn">{addOn}</span>}
                </span>
            );
        case 'sm':
            return (
                <span>
                    <h4 className="SectionHeader__title">{title}</h4>
                    {addOn && <span className="SectionHeader__addOn">{addOn}</span>}
                </span>
            );
        case 'xl':
        default:
            return (
                <span>
                    <h1 className="SectionHeader__title">{title}</h1>
                    {addOn && <span className="SectionHeader__addOn">{addOn}</span>}
                </span>
            );
    }
}

export const SectionHeader = ({ addOn, children, className, size, title }) => {
    const classes = classNames('SectionHeader', className, `size-${size}`);
    return (
        <header className={classes}>
            {getHeading(size, title, addOn)}
            <div className="SectionHeader__controls">{children}</div>
        </header>
    );
};

SectionHeader.defaultProps = defaultProps;

SectionHeader.propTypes = propTypes;
