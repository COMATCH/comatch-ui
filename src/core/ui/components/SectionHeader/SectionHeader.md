
## SectionHeader

From [`SectionHeader.js`](SectionHeader.js)



prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**addOn** | `Union<Array[]<ReactNode> \| ReactNode>` | `null` | :x: | 
**children** | `Union<Array[]<ReactNode> \| ReactNode>` | `[]` | :x: | 
**className** | `String` | `null` | :x: | 
**size** | `Enum('xs','sm','xl')` | `'xl'` | :x: | 
**title** | `Union<String \| ReactNode>` | `''` | :x: | 



