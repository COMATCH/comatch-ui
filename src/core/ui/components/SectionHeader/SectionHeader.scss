@import '../../../styles/variables.scss';

.SectionHeader {
    display: flex;
    position: relative;

    &__title {
        margin: 0;
        font-weight: bold;
        flex-grow: 1;
        display: inline;
    }

    &__addOn {
        margin-left: 5px;
    }

    &__controls {
        .Button {
            margin-left: 10px;
            &:first {
                margin-left: 0;
            }
        }
    }

    &:after {
        background-color: $co-color-mid-gray;
        content: '';
        display: block;
        height: 1px;
        width: 100%;
        position: absolute;
        left: 0;
    }

    // sizes
    &.size-xs {
        margin-bottom: 20px;

        .SectionHeader__title {
            font-size: $co-font-sm;
        }

        &:after {
            bottom: -5px;
        }
    }

    &.size-sm {
        margin-bottom: 30px;

        .SectionHeader__title {
            font-size: $co-font-md;
        }

        &:after {
            bottom: -5px;
        }
    }

    &.size-xl {
        margin-bottom: 30px;

        .SectionHeader__title {
            font-size: $co-font-xl;
            line-height: 1.3;
        }

        &:after {
            bottom: -5px;
        }
    }
}
