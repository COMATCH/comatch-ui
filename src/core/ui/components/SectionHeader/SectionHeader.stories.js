import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons/faTimes';
import { storiesOf } from '@storybook/react';
import { Button } from '../Button';
import { SectionHeader } from './SectionHeader';

storiesOf('SectionHeader', module)
    .add('with title translate', () => <SectionHeader title="Fancy title" />)
    .add('with title string', () => (
        <SectionHeader title="I am the title of the Universe and of all the other universes" />
    ))
    .add('with title and buttons', () => (
        <SectionHeader title="Test title">
            <Button iconAfterText={true} icon={<FontAwesomeIcon icon={faTimes} />} text="Hello world" />
            <Button iconAfterText={true} icon={<FontAwesomeIcon icon={faTimes} />} text="Hello world" />
        </SectionHeader>
    ));
