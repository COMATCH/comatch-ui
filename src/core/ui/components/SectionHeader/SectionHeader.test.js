import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { lorem } from 'faker';
import { SectionHeader } from './SectionHeader';
import '../../../config/tests/setup';

describe('SectionHeader tests', () => {
    const DEFAULT_PROPS = {
        className: 'SectionHeader size-xl',
    };

    const OPTIONAL_PROPS = {
        className: lorem.word(),
        title: lorem.words(),
        children: ['btn1', 'btn2', 'btn3'],
    };

    describe('given render the default SectionHeader component', () => {
        it('should render and check snapshot correctly', () => {
            const sectionHeader = renderer.create(<SectionHeader />).toJSON();
            expect(sectionHeader).toMatchSnapshot();
        });
        it('should render the default SectionHeader without crashing', () => {
            const phoneNumberInput = renderer.create(<SectionHeader />).toJSON();
            expect(phoneNumberInput).toBeTruthy();
        });
    });

    describe('given default props', () => {
        it('should render className correctly', () => {
            const sectionHeader = mount(<SectionHeader />);
            expect(sectionHeader.prop('className')).toEqual(null);
        });
    });

    describe('given default props', () => {
        const props = {
            ...DEFAULT_PROPS,
        };
        it('should render correctly with default prop', () => {
            const sectionHeader = shallow(<SectionHeader />);
            expect(sectionHeader.prop('className')).toEqual(props.className);
        });
    });

    describe('given optional props', () => {
        const props = {
            ...OPTIONAL_PROPS,
        };
        it('should render header className with `size` correctly', () => {
            const sectionHeader = shallow(<SectionHeader className={props.className} />);
            expect(sectionHeader.prop('className')).toEqual(`SectionHeader ${props.className} size-xl`);
        });

        it('should render header className with `title` correctly', () => {
            const sectionHeader = shallow(<SectionHeader />);
            expect(
                sectionHeader
                    .find('span h1')
                    .at(0)
                    .props().className,
            ).toEqual('SectionHeader__title');
        });

        it('should render header className with `controls` correctly', () => {
            const sectionHeader = shallow(<SectionHeader />);
            expect(
                sectionHeader
                    .find('div')
                    .at(0)
                    .props().className,
            ).toEqual('SectionHeader__controls');
        });
    });

    describe('given title and children props', () => {
        const props = {
            ...OPTIONAL_PROPS,
        };
        it('should render `title` prop with text correctly', () => {
            const sectionHeader = mount(<SectionHeader title={props.title} />);
            expect(sectionHeader.find('span h1').text()).toEqual(props.title);
        });
        it('should render `children` props correctly', () => {
            const sectionHeader = shallow(<SectionHeader {...props} />);
            expect(
                sectionHeader
                    .find('div')
                    .at(0)
                    .props().children,
            ).toEqual(props.children);
        });
    });
});
