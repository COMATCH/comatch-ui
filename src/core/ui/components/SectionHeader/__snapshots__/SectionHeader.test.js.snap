// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader tests given render the default SectionHeader component should render and check snapshot correctly 1`] = `
<header
  className="SectionHeader size-xl"
>
  <span>
    <h1
      className="SectionHeader__title"
    >
      
    </h1>
  </span>
  <div
    className="SectionHeader__controls"
  />
</header>
`;
